wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/OAuthAuthenticationService.java

```diff
@@ -103,8 +102,9 @@ public Response authenticate(@Required @QueryParam("oauth_provider") String oaut
 
         OAuthAuthenticator oauth = getAuthenticator(oauthProvider);
         if (!isNullOrEmpty(userId) && !userId.equals(EnvironmentContext.getCurrent().getSubject().getUserId())) {
-            throw new ForbiddenException(
-                    "Provided userId " + userId + " is not related to current user " + EnvironmentContext.getCurrent().getSubject().getUserId());
+            throw new ForbiddenException("Provided userId " + userId + " is not related to current user " + EnvironmentContext.getCurrent()
```

Comment:
You call `EnvironmentContext.getCurrent().getSubject().getUserId()` twice. Please save it to variable and reuse